### PR TITLE
Fix `required_features` for reversible transforms

### LINKED ISSUE #963: Fix `required_features` for reversible transforms
### 🚀 Feature Request

Reversible transforms also requires target quantiles to be in `required_features` list to do the correct `inverse_transform` 

### Proposal
1. Change method `inverse_transform` in `ReversibleTransform`:
- Check whether "target" in `required_features`
- If so, add target quantiles to the list if `required_features + [quantile_1, ..]` in to_pandas call

2. Make sure that all the `ReversibleTransform` pass [in_column] as `required_features`
3. Add test for the new logic: inverse_transform pass the target quantiles in to_pandas
4. Fix tests with `@pytest.mark.xfail(reason="TSDataset 2.0: bug in required_features with quantiles")`

### Test cases

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by alex-hse-repository**
## Before submitting (must do checklist)

- [ ] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs? We use Numpy format for all the methods and classes.
- [ ] Did you write any new necessary tests?
- [ ] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Closing issues

closes #963

### CODE REVIEW

File: etna/pipeline/pipeline.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ def _forecast(self) -> TSDataset:
 
         if isinstance(self.model, DeepBaseModel):
             future = self.ts.make_future(
-                future_steps=self.model.decoder_length, transforms=self.transforms, tail_steps=self.model.encoder_length
+                future_steps=self.horizon, transforms=self.transforms, tail_steps=self.model.encoder_length
```
Comment (Mr-Geekman) [inline etna/pipeline/pipeline.py:59]: Why do you fix it here? May be it is better of merge this fix from master during merging.

File: tests/test_transforms/test_base/test_base.py [hunk 2]
```diff
@@ -7,6 +7,7 @@
 from etna.datasets import TSDataset
 from etna.datasets import generate_ar_df
 from etna.transforms import IrreversibleTransform
+from etna.transforms import ReversibleTransform
 
 
 class NewTransformMock(IrreversibleTransform):
```
Comment (Mr-Geekman) [inline tests/test_transforms/test_base/test_base.py:13]: It isn't a new transform currently.

File: etna/transforms/base.py [hunk 3]
```diff
@@ -223,7 +224,11 @@ def inverse_transform(self, ts: TSDataset) -> TSDataset:
         :
             TSDataset after applying inverse transformation.
         """
-        df = ts.to_pandas(flatten=False, features=self.required_features)
+        required_features = self.required_features
+        if isinstance(required_features, list) and "target" in self.required_features:
+            features = set(ts.columns.get_level_values("feature").tolist())
+            required_features += list(match_target_quantiles(features))
```
Comment (Mr-Geekman) [inline etna/transforms/base.py:230]: Don't we want to get rid of duplicates here? What if in `required_features` we already have quantiles?

File: etna/transforms/math/sklearn.py [hunk 4]
```diff
@@ -59,15 +59,14 @@ def __init__(
...
         """
-        super().__init__(required_features="all")
-        if inplace and (out_column is not None):
-            warnings.warn("Transformation will be applied inplace, out_column param will be ignored")
-
-        self.transformer = transformer
-
         if isinstance(in_column, str):
             in_column = [in_column]
+        super().__init__(required_features=sorted(in_column) if in_column is not None else "all")
```
Comment (Mr-Geekman) [inline etna/transforms/math/sklearn.py:64]: Looks difficult for one line. Move expression in `__init__` on the line above.

**Review by Mr-Geekman — CHANGES_REQUESTED** [review]
Look at comments above.

File: etna/pipeline/pipeline.py (hunk 1 above)
Comment (alex-hse-repository) [reply etna/pipeline/pipeline.py:59]: I have recently rebased on master branch, this is a small artefact

File: tests/test_transforms/test_base/test_base.py (hunk 2 above)
Comment (alex-hse-repository) [reply tests/test_transforms/test_base/test_base.py:13]: Fixed

File: etna/transforms/base.py (hunk 3 above)
Comment (alex-hse-repository) [reply etna/transforms/base.py:230]: Fixed

File: etna/transforms/math/sklearn.py (hunk 4 above)
Comment (alex-hse-repository) [reply etna/transforms/math/sklearn.py:64]: Fixed

**Comment by Mr-Geekman** [comment]
Why do we need `tests/test_transforms/test_base/__init__.py`?

File: etna/transforms/base.py [hunk 5]
```diff
@@ -223,7 +232,7 @@ def inverse_transform(self, ts: TSDataset) -> TSDataset:
         :
             TSDataset after applying inverse transformation.
         """
-        df = ts.to_pandas(flatten=False, features=self.required_features)
+        df = ts.to_pandas(flatten=False, features=self._get_inverse_transform_required_features(ts))
         columns_before = set(df.columns.get_level_values("feature"))
         df_transformed = self._inverse_transform(df=df)
```
Comment (Mr-Geekman) [inline etna/transforms/base.py:237]: We don't test calling this.

File: etna/transforms/base.py [hunk 6]
```diff
@@ -223,7 +232,7 @@ def inverse_transform(self, ts: TSDataset) -> TSDataset:
...
         ts = self._update_dataset(ts=ts, columns_before=columns_before, df_transformed=df_transformed)
```
Comment (Mr-Geekman) [inline etna/transforms/base.py:238]: We don't test calling this.

**Review by Mr-Geekman — APPROVED** [review]

**Pull request merged**